src/modules/memory.cpp

```diff
@@ -13,13 +13,30 @@ waybar::modules::Memory::Memory(const Json::Value& config)
 
 auto waybar::modules::Memory::update() -> void
 {
-  struct sysinfo info = {};
-  if (sysinfo(&info) == 0) {
-    auto total = info.totalram * info.mem_unit;
-    auto freeram = info.freeram * info.mem_unit;
-    int used_ram_percentage = 100 * (total - freeram) / total;
-    label_.set_text(fmt::format(format_, used_ram_percentage));
-    auto used_ram_gigabytes = (total - freeram) / std::pow(1024, 3);
-    label_.set_tooltip_text(fmt::format("{:.{}f}Gb used", used_ram_gigabytes, 1));
+  parseMeminfo();
+  int used_ram_percentage = 100 * (memtotal_ - memfree_) / memtotal_;
+  label_.set_text(fmt::format(format_, used_ram_percentage));
+  auto used_ram_gigabytes = (memtotal_ - memfree_) / std::pow(1024, 2);
+  label_.set_tooltip_text(fmt::format("{:.{}f}Gb used", used_ram_gigabytes, 1));
+}
+
+void waybar::modules::Memory::parseMeminfo()
+{
+  int memtotal, memfree, memavail, membuffer, memcache;
+  FILE* info = fopen("/proc/meminfo","r");
+  if(fscanf (info, "MemTotal: %d kB MemFree: %d kB Buffers: %d kB Cached: %d kB",&memtotal, &memfree, &membuffer, &memcache) < 4) { // Old meminfo format 
+    fclose(info);
+    info = fopen("/proc/meminfo","r"); 
```

Comment:
Why close and re-open?

Reply:
You're right, after looking at it again, this can be replaced by `rewind(info);`